### ISSUE TITLE
SetTheme() behaves is not fully effective for widget.Form

### ISSUE THREAD

**Issue by BabySid** [fyne-io/fyne#2810]
### Describe the bug:
I have the same problem #842
When I SetTheme(), I found that the text color of FormItem did not refresh.
I checked my code and found that there should be no misuse in use. I searched the issue and found similar problems as above.

### To Reproduce:
Steps to reproduce the behaviour:
1. Run the code below

### Screenshots:
When Theme is Light, it is normal.
![微信图片_20220227181132](https://user-images.githubusercontent.com/6658943/155878306-6d112371-76ce-41e9-bb77-0e46979c14f6.png)
When Theme switch to Dark, I hope the text of FormItems is white, but it's actually black.
![微信图片_20220227181136](https://user-images.githubusercontent.com/6658943/155878309-9366d0a3-f033-4da2-89ff-172be3759655.png)

### Example code:
```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/theme"
	"fyne.io/fyne/v2/widget"
)

func main() {
	myApp := app.New()
	myWindow := myApp.NewWindow("Rick&Morty")
	myWindow.SetMainMenu(fyne.NewMainMenu(fyne.NewMenu("Menu",
		fyne.NewMenuItem("Dark Theme", func() { myApp.Settings().SetTheme(theme.DarkTheme()) }),
		fyne.NewMenuItem("Light Theme", func() { myApp.Settings().SetTheme(theme.LightTheme()) }),
	)))

	form := widget.NewForm()
	form.Append("ID", widget.NewLabel("Character's Name"))
	form.Append("be", widget.NewLabel("bebe"))

	myWindow.SetContent(form)
	myWindow.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Windows10
 - **Version:** 19044
 - **Go version:** 1.17.5
 - **Fyne version:** 2.1.3

**Comment by andydotxyz** [comment]
Thanks for reporting this, confirmed

**Comment by andydotxyz** [comment]
Resolved for next release (v2.1.4 or v2.2.0)
